js/components/eiti-data-map.js

```diff
@@ -87,46 +87,50 @@
           var svgLegend = d3.select(this)
             .select('.legend-svg');
 
-          svgLegend.append('g')
-            .attr('class', 'legendScale');
-
-          var legend = d3.legend.color()
-            .labelFormat(format.si)
-            .useClass(false)
-            .ascending(true)
-            .labelDelimiter('-')
-            .shapePadding(6)
-            .scale(scale);
-
-          svgLegend.select('.legendScale')
-            .call(legend);
-
-          // reverse because the scale is in ascending order
-          var _steps = d3.range(0, 9).reverse();
-
-          // find which steps are represented in the map
-          var uniqueSteps = getUnique(marks.data(), _steps, domain);
-
-          // start consolidate (translate) visible cells
-          var cells = svgLegend.selectAll('.cell');
-          var cellHeight = legend.shapeHeight() + legend.shapePadding();
-          var count = 0;
-          cells.each(function(cell, i) {
-            var present = uniqueSteps.indexOf(i) > -1;
-
-            if (!present) {
-              // hide cells swatches that aren't in the map
-              cells[0][i].setAttribute('aria-hidden', true);
-              count++;
-            } else  {
-              // trim spacing between swatches that are visible
-              var translateHeight = (i * cellHeight) - (count * cellHeight);
-              cells[0][i].setAttribute('transform',
-                'translate(0,' + translateHeight + ')');
-            }
-          });
-          // end consolidation
-          // end map legend
+          if (!svgLegend.empty()) {
+            svgLegend.append('g')
+              .attr('class', 'legendScale');
+
+            var legend = d3.legend.color()
+              .labelFormat(format.si)
+              .useClass(false)
+              .ascending(true)
+              .labelDelimiter('-')
+              .shapePadding(6)
+              .scale(scale);
+
+            svgLegend.select('.legendScale')
+              .call(legend);
+
+            // reverse because the scale is in ascending order
+            var _steps = d3.range(0, 9).reverse();
```

Comment:
I think I did this to not overwrite `steps` from above, but I guess I'm not totally sure that would be problematic, come to think of it. I merged for now, because this came up in standup, but should be refactored later.